sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonSchemaReader.java

```diff
@@ -283,6 +280,96 @@ public static Schema readSchemaInIndexFile(String indexFilePath) throws IOExcept
     return readSchema(indexFilePath, false);
   }
 
+  public static StructField getStructChildren(CarbonTable table, String columnName) {
+    List<CarbonDimension> list = table.getChildren(columnName);
+    List<StructField> structFields = new ArrayList<StructField>();
+    for (int i = 0; i < list.size(); i++) {
+      CarbonDimension carbonDimension = list.get(i);
```

Comment:
Can you extract common code from `getStructChildren` and `getArrayChildren`methods to new method and reuse

Reply:
done